File: python/hardware/storage/hdd.py
from datetime import date
from hardware.storage.storage_device import StorageDevice

class HDD(StorageDevice):
    def __init__(self, serial_number="", installation_date=None, physically_installed=False, capacity_gb=0, read_speed_mbs=0, write_speed_mbs=0, rpm=0, platter_count=0, id="", manufacturer="", model="", price=0.0):
        super().__init__(serial_number, installation_date, physically_installed, capacity_gb, read_speed_mbs, write_speed_mbs, id, manufacturer, model, price)
        self._rpm = rpm
        self._platter_count = platter_count

    # Setters
    def set_rpm(self, rpm):
        self._rpm = rpm

    def set_platter_count(self, platter_count):
        self._platter_count = platter_count

    # Getters
    def get_rpm(self):
        return self._rpm

    def get_platter_count(self):
        return self._platter_count

    def connect(self):
        print("HDD connected")
        return True

    def disconnect(self):
        print("HDD disconnected")
        return True

File: cpp/hardware/storage/HDD.cpp
#pragma once
#include <bits/stdc++.h>

#include "StorageDevice.cpp"

class HDD : public StorageDevice
{
private:
    int rpm;
    int platterCount;

public:
    // Constructor
    HDD() : StorageDevice(), rpm(0), platterCount(0) {}
    HDD(const std::string &serial, const Date &installDate, bool installed, int capacityGB, int readSpeedMBs, int writeSpeedMBs, int rpm, int platterCount, const std::string &id, const std::string &manufacturer, const std::string &model, double price)
        : StorageDevice(serial, installDate, installed, capacityGB, readSpeedMBs, writeSpeedMBs, id, manufacturer, model, price), rpm(rpm), platterCount(platterCount) {}

    // Destructor
    ~HDD() {}

    // Setters
    void setRpm(int rpm)
    {
        this->rpm = rpm;
    }

    void setPlatterCount(int platterCount)
    {
        this->platterCount = platterCount;
    }

    // Getters
    int getRpm() const
    {
        return rpm;
    }

    int getPlatterCount() const
    {
        return platterCount;
    }

    bool connect() override
    {
        printf("HDD connected\n");
        return true;
    }

    bool disconnect() override
    {
        printf("HDD disconnected\n");
        return true;
    }
};